src/anomalib/data/base/datamodule.py

```diff
@@ -155,66 +176,143 @@ def _setup(self, _stage: str | None = None) -> None:
+                split_ratio = 0.5
+            else:
+                split_ratio = self.val_split_ratio
+
+            self.val_data, self.test_data = self.test_data.create_subset(
+                criteria=[split_ratio, 1 - split_ratio],
                 seed=self.seed,
-            )
-        elif self.val_split_mode == ValSplitMode.FROM_TEST:
-            # randomly sampled from test set
-            self.test_data, self.val_data = random_split(
-                self.test_data,
-                self.val_split_ratio,
                 label_aware=True,
+            )
+        elif self.val_split_mode == SplitMode.PREDEFINED:
+            logger.warning(
+                "Skipping val set creation. This means that the model will not be evaluated."
+                "You can use 'SplitMode.AUTO' to automatically create a val set, "
+                "or 'SplitMode.SYNTHETIC' to generate synthetic val set.",
+            )
+        elif self.val_split_mode == SplitMode.SYNTHETIC:
+            logger.info("Generating synthetic val set.")
+            self.val_data = SyntheticAnomalyDataset.from_dataset(self.train_data)
+        else:
+            msg = f"Invalid val split mode: {self.val_split_mode}"
+            raise ValueError(msg)
+
+    def _process_train_val_scenario(self) -> None:
+        if self.test_split_mode == SplitMode.AUTO:
+            if self.test_split_ratio is None:
+                logger.info("'test_split_ratio' is not specified. Choosing a default value of 0.5 for AUTO mode.")
+                split_ratio = 0.5
+            else:
+                split_ratio = self.test_split_ratio
+
+            self.val_data, self.test_data = self.val_data.create_subset(
+                criteria=[1 - split_ratio, split_ratio],
                 seed=self.seed,
+                label_aware=True,
             )
-        elif self.val_split_mode == ValSplitMode.SAME_AS_TEST:
-            # equal to test set
-            self.val_data = self.test_data
-        elif self.val_split_mode == ValSplitMode.SYNTHETIC:
-            # converted from random training sample
-            self.train_data, normal_val_data = random_split(self.train_data, self.val_split_ratio, seed=self.seed)
-            self.val_data = SyntheticAnomalyDataset.from_dataset(normal_val_data)
-        elif self.val_split_mode != ValSplitMode.NONE:
-            msg = f"Unknown validation split mode: {self.val_split_mode}"
+        elif self.test_split_mode == SplitMode.PREDEFINED:
+            logger.warning(
+                "Skipping test set creation. This means that the model will not be evaluated."
+                "You can use 'SplitMode.AUTO' to automatically create a test set, "
+                "or 'SplitMode.SYNTHETIC' to generate synthetic test set.",
+            )
+        elif self.test_split_mode == SplitMode.SYNTHETIC:
+            logger.info("Generating synthetic test set.")
+            self.test_data = SyntheticAnomalyDataset.from_dataset(self.train_data)
```

Comment:
Same as above. We need to split the train set first, to ensure that the train and test sets are mutually exclusive.